QuantLib/ql/termstructures/yield/nonlinearfittingmethods.cpp

```diff
@@ -213,6 +222,35 @@ namespace QuantLib {
         }
         return d;
     }
+	
+	SpreadFittingMethod::SpreadFittingMethod(shared_ptr<FittingMethod> method,
+                        Handle<YieldTermStructure> discountCurve)
+    : FittedBondDiscountCurve::FittingMethod(method ? method->constrainAtZero() : true, method ? method->weights() : Array(), 
+											 method ? method->optimizationMethod() : boost::shared_ptr<OptimizationMethod>()),
+      method_(method), discountingCurve_(discountCurve) {
+		QL_REQUIRE(!method, "Fitting method is empty");
```

Comment:
Sorry, I didn't test it after the changes. Let me see...

On Fri, 23 Oct 2015 16:35 Luigi Ballabio <notifications@github.com> wrote:

> In QuantLib/ql/termstructures/yield/nonlinearfittingmethods.cpp
> <https://github.com/lballabio/quantlib/pull/288#discussion_r42871693>:
>
> > @@ -213,6 +222,35 @@ namespace QuantLib {
> >          }
> >          return d;
> >      }
> > +	
> > +	SpreadFittingMethod::SpreadFittingMethod(shared_ptr<FittingMethod> method,
> > +                        Handle<YieldTermStructure> discountCurve)
> > +    : FittedBondDiscountCurve::FittingMethod(method ? method->constrainAtZero() : true, method ? method->weights() : Array(),
> > +											 method ? method->optimizationMethod() : boost::shared_ptr<OptimizationMethod>()),
> > +      method_(method), discountingCurve_(discountCurve) {
> > +		QL_REQUIRE(!method, "Fitting method is empty");
>
> You're negating method, so this actually requires that the method is
> null. It should be QL_REQUIRE(method, ...) instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lballabio/quantlib/pull/288/files#r42871693>.
>